tests/functional/includes/date.mock.js

```diff
@@ -0,0 +1,14 @@
+// Highjack the Date object so that our tests don't suddenly start failing...
```

Comment:
I created this mock object to guarantee that the code always returns predictable dates. Otherwise, testing becomes impossible. I could not think of a better way. Note that I don't temper with the original Date object that much. I just make sure that `new Date()` (called without a parameter) and `Date.now()` return a specific value. Everything else remains the same.